teloc_macros/src/lib.rs

```diff
@@ -82,3 +82,23 @@ pub fn inject(_: TokenStream, input: TokenStream) -> TokenStream {
     let tokens = res.unwrap_or_else(identity);
     (quote::quote! { #imp #tokens }).into()
 }
+
+/// This macro informs `#[inject]` which method in an impl block is used to create the type.
+///
+/// Example:
+/// ```compile_fail
+/// use teloc::{inject, inject_init};
+///
+/// struct Number(i8);
+/// #[inject]
+/// impl Number {
+///     #[inject_init]
+///     fn init(number: u8) -> Self { Number(number) }
+///
+///     fn negate(number: u8) -> Self { Number(-number) }
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn inject_init(_: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
Ideally, parsing`#[inject]` macro would delete this macro (it would be named `#[inject::init]` to imitate a qualified name lookup), but that seems like a much more complicated change. Lmk if you think it's worthwhile implementing that

Reply:
`#[inject::init]` seems more (rusty?) than `#[inject_init]`, but it's not too important. If you have time, you can implement it.
We can also change `inject_init` name to a simple `#[init]` to be more neat.